presto-prometheus/src/main/java/com/facebook/presto/plugin/prometheus/PrometheusRecordCursor.java

```
@@ -0,0 +1,302 @@
+        }
+        fields = metricsItr.next();
+        return true;
+    }
+
+    @Override
+    public boolean getBoolean(int field)
+    {
+        return true;
+    }
+
+    @Override
+    public long getLong(int field)
+    {
+        Type type = getType(field);
+        if (type.equals(TIMESTAMP_COLUMN_TYPE)) {
+            Instant dateTime = (Instant) requireNonNull(getFieldValue(field));
+            // render with the fixed offset of the Presto server
+            int offsetMinutes = dateTime.atZone(ZoneId.systemDefault()).getOffset().getTotalSeconds() / 60;
+            return packDateTimeWithZone(dateTime.toEpochMilli(), offsetMinutes);
+        }
+        else {
+            throw new PrestoException(NOT_SUPPORTED, "Unsupported type " + getType(field));
+        }
+    }
+
+    @Override
+    public double getDouble(int field)
+    {
+        checkFieldType(field, DOUBLE);
+        return (double) requireNonNull(getFieldValue(field));
+    }
+
+    @Override
+    public Slice getSlice(int field)
+    {
+        checkFieldType(field, createUnboundedVarcharType());
+        return Slices.utf8Slice((String) requireNonNull(getFieldValue(field)));
+    }
+
+    @Override
+    public Object getObject(int field)
+    {
+        return getFieldValue(field);
+    }
+
+    @Override
+    public boolean isNull(int field)
+    {
+        checkArgument(field < columnHandles.size(), "Invalid field index");
+        if (getFieldValue(field) == null) {
+            return true;
+        }
+        return false;
+    }
+
+    private Object getFieldValue(int field)
+    {
+        checkState(fields != null, "Cursor has not been advanced yet");
+
+        int columnIndex = fieldToColumnIndex[field];
+        switch (columnIndex) {
+            case 0:
+                return fields.getLabels();
+            case 1:
+                return fields.getTimestamp();
+            case 2:
+                return fields.getValue();
+        }
+        return null;
```

let's throw exception or add checks, instead of return null here